wordpress-mobile/WordPress-Android
# Add automated play store uploads

I am targeting this PR against the release branch since it doesn't change anything in the actual code and it makes it much easier to test. I also need to release a new alpha/beta, but since it is for a very minor change, I wanted to use the opportunity to test the PR.

P.S: I suggest reviewing this PR with the [Hide whitespace changes](https://cloudup.com/c9eOMwgSPYL) option enabled. I committed the whitespace changes my editor made to make it easier to work with the file in the future.

**To test:**
~1. In Play Store discard the drafts for both alpha and beta and remove the artifacts for `883` and `884`. These were added with the script, so I wanted to keep it as a test instruction.~ **Edit:** I removed these, so you can ignore this step.
2. Disable [this pre-check](https://github.com/wordpress-mobile/WordPress-Android/blob/add/automated-play-store-uploads/fastlane/Fastfile#L230-L233) since it'll prevent the script running on a branch that's not a `release` branch.
3. Run `bundle exec fastlane run configure_apply` which will download the upload credentials
4. Run `bundle exec fastlane build_and_upload_pre_releases`: This will build and upload alpha and beta builds we deleted in `Step 1`.

No cleanup is necessary since we want to release a new beta and if it turns out to be necessary, I'll take care of it. Also, this will not test the `build_and_upload_release` lane, but it's pretty much the same as the pre-releases with only difference being the `production` track.

**Edit:** Cleanup will be necessary since there is a new PR against the release branch, but I can do that.

**PR submission checklist:**
